### PR TITLE
Updating OpenZeppelin tutorial to use Truffle 4

### PR DESCRIPTION
**Pull request by bmmpxf**
@DiscRiskandBisque Tutorial now uses Truffle Develop. I also updated the screenshots. Let me know what you think!

### CODE REVIEW

File: public/tutorials/robust-smart-contracts-with-openzeppelin.md [hunk 1]
```diff
@@ -112,51 +114,102 @@ Using less than 15 lines of hand-coded Solidity, we've created our own Ethereum
 
    The `import` statement within our `TutorialToken` contract will be automatically handled by the compiler, along with any subsequent imports within `StandardToken`.
 
-1. Now we can compile and migrate. In a second terminal window, run the TestRPC:
+1. Now we are ready to compile and migrate to the blockchain. We will do this using Truffle Develop, a development console that includes an in-memory blockchain, similar to the [EthereumJS TestRPC](https://github.com/ethereumjs/testrpc). In your terminal, make sure you are in the root of your project's directory and then launch Truffle Develop:
```
Comment (tcoulter) [inline public/tutorials/robust-smart-contracts-with-openzeppelin.md]: "migrating to the blockchain" sounds weird in this context. Migrations deploy your contracts to the blockchain; deploying is the goal, and "running migrations" is the action (i.e., the means to an end). Here, I'd either use "deploy your contracts to the blockchain" or "run your migrations". Obviously the latter is confusing if you haven't introduced migrations yet.

File: public/tutorials/robust-smart-contracts-with-openzeppelin.md [hunk 2]
```diff
@@ -112,51 +114,102 @@ Using less than 15 lines of hand-coded Solidity, we've created our own Ethereum
...
 
    ```shell
-   testrpc
+   truffle develop
    ```
 
-1. Back in the first terminal window, run the following commands to compile and migrate the contract to the TestRPC:
+   <p class="alert alert-info">
+     <strong>Note</strong>: If you're on Windows and encountering problems running this command, please see the documentation on [resolving naming conflicts on Windows](/docs/advanced/configuration#resolving-naming-conflicts-on-windows).
+   </p>
```
Comment (tcoulter) [inline public/tutorials/robust-smart-contracts-with-openzeppelin.md]: Awesome addition.

File: public/tutorials/robust-smart-contracts-with-openzeppelin.md [hunk 3]
```diff
@@ -112,51 +114,102 @@ Using less than 15 lines of hand-coded Solidity, we've created our own Ethereum
...
+
+1. Once the Truffle Develop prompt displays, run the following command to compile the contract:
 
    ```shell
-   truffle compile
-   truffle migrate
+   compile
+   ```
+
+1. Once the compile has completed, migrate the contract to the blockchain:
```
Comment (tcoulter) [inline public/tutorials/robust-smart-contracts-with-openzeppelin.md]: Same here for "migrate" language.

File: public/tutorials/robust-smart-contracts-with-openzeppelin.md [hunk 4]
```diff
@@ -112,51 +114,102 @@ Using less than 15 lines of hand-coded Solidity, we've created our own Ethereum
...
+
+   ```shell
+   migrate
    ```
 
    <p class="alert alert-info">
-   <strong>Note</strong>: If you're on Windows and encountering problems running these commands, please see the documentation on [resolving naming conflicts on Windows](/docs/advanced/configuration#resolving-naming-conflicts-on-windows).
+     <strong>Note</strong>: As an alternative to using Truffle Develop, both the `compile` and `migrate` commands can be run directly on the command line with `truffle compile` and `truffle migrate` respectively.
```
Comment (tcoulter) [inline public/tutorials/robust-smart-contracts-with-openzeppelin.md]: This sentence is misleading (I believe; @gnidan, correct me if I'm wrong here). `truffle migrate` won't use the `truffle develop` blockchain unless you have `truffle develop` already running and the command running inside the console provided by `truffle develop`. `truffle compile` doesn't need a blockchain environment as it doesn't talk to the blockchain, but `truffle migrate` does, and the only way it knows about the `develop` blockchain is if its within a `truffle develop` console.

File: public/tutorials/robust-smart-contracts-with-openzeppelin.md [hunk 5]
```diff
@@ -112,51 +114,102 @@ Using less than 15 lines of hand-coded Solidity, we've created our own Ethereum
...
+
+   ![TutorialToken Wallet transfer recipient](/tutorials/images/open-zeppelin/oz-transfer-address.png)
+
+   *TutorialToken wallet transfer recipient*
+
+   <p class="alert alert-info">
+     <strong>Note</strong>: To review the list of accounts, type `web3.eth.accounts` in Truffle Develop.
+   </p>
+
+1. Click "Transfer" to initiate the token transfer. Metamask will intercept the transfer request and display a confirmation. Note that no ether is changing hands, except in the form of gas.
```
Comment (tcoulter) [inline public/tutorials/robust-smart-contracts-with-openzeppelin.md]: I'd probably change the tail end of this sentence to say "except for the gas used to pay for the transaction." Something to suggest the point of the gas.

File: public/tutorials/robust-smart-contracts-with-openzeppelin.md (hunk 4 above)
Comment (gnidan) [reply public/tutorials/robust-smart-contracts-with-openzeppelin.md]: This is all correct – `truffle migrate` has no way to use the Truffle Develop blockchain, unless invoked as `migrate` from within the Develop console

File: public/tutorials/robust-smart-contracts-with-openzeppelin.md (hunk 1 above)
Comment (bmmpxf) [reply public/tutorials/robust-smart-contracts-with-openzeppelin.md]: Ah I see. I confess I still get a little confused between "deploy" and "migrate" and was using them interchangeably in this case. Will update to say "deploy your contract to the blockchain" (there is only one, correct?).

File: public/tutorials/robust-smart-contracts-with-openzeppelin.md (hunk 4 above)
Comment (bmmpxf) [reply public/tutorials/robust-smart-contracts-with-openzeppelin.md]: Right, of course, I neglected to mention there that you would have to use those commands *while connected to some other blockchain* (such as testrpc). Didn't mean to imply that Truffle Develop would know about it. Will update.

File: public/tutorials/robust-smart-contracts-with-openzeppelin.md (hunk 5 above)
Comment (bmmpxf) [reply public/tutorials/robust-smart-contracts-with-openzeppelin.md]: Good point, that's clearer.

**Comment by bmmpxf** [comment]
Thanks for the feedback @tcoulter! I just pushed a change that should resolve all those issues.

**Comment by bmmpxf** [comment]
Got the :+1: from @tcoulter and merging in.

**Pull request merged**